les/peer.go

```diff
@@ -948,8 +949,18 @@ func (p *clientPeer) freezeClient() {
 // Handshake executes the les protocol handshake, negotiating version number,
 // network IDs, difficulties, head and genesis blocks.
 func (p *clientPeer) Handshake(td *big.Int, head common.Hash, headNum uint64, genesis common.Hash, server *LesServer) error {
-	disableTxLookup := server.handler.blockchain.TxLookupLimit() != 0
-	if disableTxLookup && p.version < lpv4 {
+	recentTx := server.handler.blockchain.TxLookupLimit()
+	if recentTx > 0 { // 0 means no limit (all txs available)
```

Comment:
Can we put these number in the protocol file and use the constants? It's a bit weird to hardcode the number here

Reply:
done

Reply:
Sorry one more nitpicks, can we replace the `1` with a constant? It's better to define in the protocol that `1` means disabled.

Reply:
done